### PR TITLE
[14.0][IMP] web_company_color: Forward port #1903 from 12.0

### PR DESCRIPTION
**Pull request by HekkiMelody**
Forward port of https://github.com/OCA/web/pull/1903 
commit e18209c

### CODE REVIEW

File: web_company_color/models/res_company.py [hunk 1]
```diff
@@ -73,7 +124,32 @@ def write(self, values):
                 "color_navbar_bg",
                 "color_navbar_bg_hover",
                 "color_navbar_text",
+                "color_button_bg",
+                "color_button_bg_hover",
+                "color_button_text",
+                "color_link_text",
+                "color_link_text_hover",
             )
+            if "logo" in values:
```
Comment (StefanRijnhart) [inline web_company_color/models/res_company.py]: You seem to be undoing https://github.com/OCA/web/pull/2158/files. Is that on purpose?

Comment (HekkiMelody) [reply web_company_color/models/res_company.py]: Ah, good catch. I didn't really follow this module's history and just ported the commit trying to preserve as much of it as possible.

Thank you :pray: 
Fixing it now.

**Comment by HekkiMelody** [comment]
Added a commit to fix #2248

**Review by StefanRijnhart — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
@hbrunn your improvement is being forward ported! Care to review?

**Review by hbrunn — APPROVED** [review]

**Review by LoisRForgeFlow — COMMENTED** [review]
I'm late to this, but I just hit into this change. I see 2 problems:

1. Starting with the field in blank you need to set all fields in order for it to have an effect. (you cannot just set the navbar colors for instance).
2. When computing colors from logo, only the nabvar ones are filled.

**Pull request merged**